src/entryproc.cpp

```diff
@@ -175,8 +177,9 @@ bool EntryProcessor::safeOpen() {
                 if (errno != ENOENT) {
                     std::cerr << "warning: skipping " << m_path << ": " << pathfd.path() << ": openat(): "
                         << std::strerror(errno) << std::endl;
+                    return false;
                 }
-                return false;
+                return true;
```

Comment:
I'm confused here a bit. Why silently return true? Is there a situation where we can get this far on a non-existing file?

Reply:
Yes, because `errno == ENOENT`, maybe I should put it in an `else` clause instead, or generally refactor the `if`.-

Reply:
The problem right now is that it silently returns `false`, which is why we don't see any additional error diagnostic in the OBS buildlog. Instead we need to allow the caller to differentiate between fatal and non fatal errors. I also thought about introducing a state `enum` that better reflects the outcome of `safeOpen()` instead of coupling it with the status of `m_fd`.

Reply:
It could be the right moment to add coherent error handling to cover all cases. It depends how long it would take, and how long this submission can wait.

Reply:
I don't think any case is missing currently, it's just not straightforward to see. I will try something using an `enum`.

Reply:
Ok yes the additional commit makes sense.